src/problems/bvp_problems.jl

```diff
@@ -119,25 +119,25 @@ struct BVProblem{uType, tType, isinplace, P, F, BF, PT, K} <:
         else
             @assert prob_type === problem_type "This indicates incorrect problem type specification! Users should never pass in `problem_type` kwarg, this exists exclusively for internal use."
         end
-        return new{typeof(u0), typeof(_tspan), iip, typeof(p), typeof(f), typeof(bc),
-            typeof(problem_type), typeof(kwargs)}(f, bc, u0, _tspan, p, problem_type,
+        return new{typeof(u0), typeof(_tspan), iip, typeof(p), typeof(f), typeof(f.bc),
```

Comment:
Why is the bc out here at all? Isn't it in the BVPFunction?

Reply:
@avik-pal is it actually needed?

Reply:
The `bc` was kept outside similar to the SDEProblem interface carrying around the `g` outside (I think?)

Reply:
That was just fixed in the v2.0. I thought this was fixed in the v2.0 as well 😅. Fix and slip it in ASAP.

Reply:
The  modifications in ```BVProblem``` is similiar to ```SDEProblem```, just try to make them consistent

Reply:
SDEProblem is still carrying around `g` https://github.com/SciML/SciMLBase.jl/blob/75b192581832ae89231df0d51a0fe9624e61470e/src/problems/sde_problems.jl#L88-L89

Reply:
Yeah, ```SDEProblem``` is still carrying ```g```, so in this way I think we can still use ```prob.g``` to directly access ```g```